internal/config/main.go

```diff
@@ -64,6 +65,7 @@ func New(opts ...Option) Config {
 		version:                       version.Get(),
 		autoscalingVersion:            autodetect.DefaultAutoscalingVersion,
 		onPlatformChange:              newOnChange(),
+		openShiftRoutesAvailable:      autodetect.DefaultOpenShiftAvailability,
```

Comment:
Could we please add a test that verifies that `openShiftRoutesAvailable` is updated in all configuration copies? Since the detection only runs on the main configuration.

Reply:
I'll take a look :)

Reply:
Does it make sense to test this? Because, if the `autodetect` is called just for the main configuration, why would this field change?

Reply:
> The same case applies for autoscalingVersion, right?

Seems like, yes. :)

> Does it make sense to test this? Because, if the autodetect is called just for the main configuration, why would this field change?

From my point of view it does. Its exported and highly misleading if it gets called from somewhere else in the future. We had that issue with `onPlatformChange` previously. It was super annoying.